### ISSUE TITLE
Default Admin password

### ISSUE THREAD

**Issue by snoski3** [Snipa22/nodejs-pool#200]
If I follow the path in the readme, to /#/admin, it just goes back to home.  If I go to admin.html, it pulls up the admin login page.  I use the default login but it says that it is invalid.

Thanks,

**Comment by ArqTras** [comment]
Add new administrator using mysql command line

**Comment by milesgratz** [comment]
I think there are some gaps & inconsistencies in the documentation, but here is an answer to your question @snoski3 - 

### How to check all Administrators in SQL - 

```
sudo mysql -u root --password
mysql> use pool;
mysql> SELECT * FROM users WHERE admin=1;
```

![image](https://user-images.githubusercontent.com/17114534/33918998-f78cebbe-df7b-11e7-93d3-a3b5c86a32b0.png)

### How to change your current Administrator password in SQL - 

```
sudo mysql -u root --password
mysql> use pool;
mysql> UPDATE users SET pass='P@ssw0rd123' WHERE username='Administrator';
```
![image](https://user-images.githubusercontent.com/17114534/33919313-a2baca78-df7d-11e7-823f-069b9013bd4f.png)

**Comment by Snipa22** [comment]
The documentation is extremely clear about this:

UPDATE pool.users SET email='your new password here' WHERE username='Administrator';

Until a password is set properly on the account, which is to say, is hashed into place, the only valid field is e-mail, not pass.

**Comment by snoski3** [comment]
I ran the command that milesgratz sent:
select * from users where admin=1;
The output is:
Table 'pool.users' doesn't exist

So I can't update a table that doesn't exist.  How do I create the users table?

**Comment by milesgratz** [comment]
Hi @snoski3  - 

Snipa22 is correct, it is documented in the [readme](https://github.com/Snipa22/nodejs-pool/blob/master/README.md). Sounds like your MySQL database is misconfigured. Check and see if the `pool` database exists and what tables exist within it.

`mysql> show databases;`

Assuming the `pool` database exists, check what tables exist...

`mysql> show tables in pool;`

**Comment by mtompkins** [comment]
I have to agree with the need for some clarity here. I've updated the pwd directly through SQL but also get the redirect to /#/home

**Comment by milesgratz** [comment]
@Snipa22  I am having the same issue as @mtompkins with `/#/admin` page. I am rebuilding again and going to use separate Ubuntu 16.04 VMs for the frontend (poolui) and the backend (api + nodejs-pool) similar to your xmrpool.net example, but unclear what the issue is... you can access `admin.html` but it doesn't respect any of the logins in the `pool.users` that are `admin=1`

**Comment by mtompkins** [comment]
@milesgratz post was helpful. If I supply the entire URL as

https://mon.gulfcoastmining.com/admin.html

I am given the page. Of note, I did login prior to that URL submission using the `Administrator` and `passwordSetInSQL`

**Comment by Snipa22** [comment]
Per the instructions @mtompkins and @milesgratz the /#/admin page isn't active:
Hop into the web interface (Should be at http://<your server IP>/#/admin), then login with Administrator/Password123, MAKE SURE TO CHANGE THIS PASSWORD ONCE YOU LOGIN. <- This step is currently not active, we're waiting for the frontend to catch up! Head down to the Manual SQL Configuration to take a look at what needs to be done by hand for now.

The /admin.html URI isn't fully working either, which is why this hasn't been updated.

**Comment by mtompkins** [comment]
@Snipa22 Thanks for the clarity (and sharing your project). Unfortunately with the current choice of language is unclear as to whether the "admin page" isn't active or the manner in which to change a password isn't implemented, "This step is currently not active" appears to refer to the directive immediately preceding it - "MAKE SURE ..." and thus the confusion.

**Comment by milesgratz** [comment]
@Snipa22  Thanks for clarifying - I misinterpreted the instructions as well.

**Comment by milesgratz** [comment]
@Snipa22  Do you think there would be some value in moving the unfinished components to a "to-do" section and keeping the instructions sequential? I think it would clear up some of the confusion.

#45 
#94 
#107 
#135

**Comment by Snipa22** [comment]
@milesgratz Personally, I don't care either way.  The instructions make perfectly good sense to me, "This bit isn't working, use the alternate instructions below".  At this point, it's unlikely that the UI will be continued to be worked on heavily, as the original developer (Mesh00) hasn't been around for > 8 months.

**Comment by snoski3** [comment]
I just closed issue #195 because last night I setup the database, then started the process from scratch again. That fixed the database issue. Now I need to go through the username steps and see if I can login to the front end.

**Comment by wenewzhang** [comment]
i stuck on this problem for one day,
http://xmr.z9p.net/#/admin while redirect to #/home,
mysql connected fine,lmdb get a lots of dead thread of reading.
can some people help me?

Thanks in advance

Thanks Snipa22

mysql> select * from pool.users;
+----+---------------+------+-----------------+-------+------------------+--------------+
| id | username      | pass | email           | admin | payout_threshold | enable_email |
+----+---------------+------+-----------------+-------+------------------+--------------+
|  5 | Administrator | NULL | guest@gmail.com |     1 |                0 |            1 |
+----+---------------+------+-----------------+-------+------------------+--------------+
1 row in set (0.00 sec)

Environment Info
  Map address: (nil)
  Map size: 25769803776
  Page size: 4096
  Max pages: 6291456
  Number of pages used: 54902
  Last transaction ID: 18185
  Max readers: 512
  Number of readers used: 32
Reader Table Status
    pid     thread     txnid
     11076 7f81670a4740 11968
     10195 7fdab1748740 12198
      4405 7f6c91397740 12444
     18536 7f6a65536740 12659
     30191 7f6b7cc5a740 12862
      5448 7f6b984d1740 13070
     17045 7ffae3a76740 13270
     31608 7f9538b07740 13476
     23185 7f9ad85bb740 13682
     15495 7feefc1d0740 13920
      1921 7f91608e7740 14116
     11292 7f9d88cb6740 14323
     27908 7f3ee5bf4740 14523
      7503 7fda52094740 14721
     16119 7f79b66d6740 14920
      2527 7f1825c14740 15126
     10910 7f6950897740 15314
     18329 7f017ffff740 15511
     31045 7f97101d5740 15725
     19474 7f46684a6740 15945
      1497 7f6ba074c740 16151
      6211 7faee4739740 16364
     18053 7fc5c179a740 16578
      8753 7f6e445c7740 16812
      1279 7fbd0ebb1740 17058
     19992 7f8a9524e740 17257
     26391 7fa921eba740 17453
      7861 7faef88d7740 17658
     13258 7f3c29a08740 17866
     26556 7f5ea559b740 18071
Freelist Status
  Tree depth: 2
  Branch pages: 1
  Leaf pages: 154

**Comment by snoski3** [comment]
@mtompkins 
Are you running an Ubuntu server?  I'm jealous.

**Comment by wenewzhang** [comment]
@snoski3  i use ubuntu server 16.04 too,just because this problem,i use the old note-cryptonode-pool instead!maybe i will fix it when i have time!

**Comment by andyjk15** [comment]
I'm getting this issue as well. It just redirects me to /home. Is this going to be a fix or should I just configure in the server?

**Comment by snoski3** [comment]
@andyjk15 
Changing the password right away worked, also launch the last nodejs modules before you try to login.

**Comment by Chinsusu** [comment]
I have a problem that I login into admin.html, it says Please check your login details.
I have changed sample_config.sql, run the sql command, but nothing else

**Comment by userid2018** [comment]
Hello snoski3 ,

I have been trying to login into the admin  page using Administrator / Password123 but i keep on getting "Please check your login details" . to be specific i have been entering "Administrator" in  "Payment Address" field and "Password123" in Password field but i still keep on getting error. Is there any step I am missing . My installer script ran successfully without any issue.

The Admin page i am trying to access is at  <<remote IP>>//admin.html#/login

here is the query result from the database as well.

mysql> select * from pool.users;
+----+---------------+------+-------------+-------+------------------+--------------+
| id | username      | pass | email       | admin | payout_threshold | enable_email |
+----+---------------+------+-------------+-------+------------------+--------------+
|  1 | Administrator | NULL | Password123 |     1 |                0 |            1 |
+----+---------------+------+-------------+-------+------------------+--------------+

**Comment by userid2018** [comment]
@snoski3 
I did a clean install . changed password in the database . updated config.json to rference public ip of Azure VM but i am still  getting the same issue where login is failing.

Also i was trying to access  http://IP:8000/leafApi . I am getting error   Cannot GET /leafApi .  Is there any configuration which needs to be changed ?
Also it will be helpful if you can point me  where log files are getting created ?

**Comment by Chinsusu** [comment]
ok, problem is https in global.js. change it into http and everything is ok

**Comment by userid2018** [comment]
@ Chinsusu,
Thanks I will try it. What do i need to make sure that http://IP:8000/leafApi doesn't  throw error  Cannot GET /leafApi .

Is there any configurations i need to change for it ?

**Comment by minereobot1** [comment]
hey buddy i have the same problem ... not have way to fix and setup the password administration during installation ?
